lesson16.md

```diff
@@ -0,0 +1,451 @@
+<!-- .slide: id="lesson16" -->
+
+# Basic Frontend - Spring 2021
+
+Lesson 16, Thursday, 2021-04-29
+
+---
+
+<!-- .slide: id="loops" -->
+
+## Loops
+
+---
+
+### Lesson overview
+
+- examples of loops
+- while loop
+- for loop
+
+---
+
+### Loops
+
+We use loops to run the same code in our programs repeatedly.
+
+More generally, a loop is a way to repeat an action multiple times as long as a certain condition is `true`.
+
+---
+
+### Loops: cake recipe 
+
+Let's say I want to bake a cake:
+
+<div style="display: flex;">
+
+![scale](images/lesson-16-scale.webp) <!-- .element: style="width: 400px"-->
+
+<div>
+Cake recipe:<br>
+
+- 2 eggs
+- 100g sugar
+- 175g flour
+- etc.
+</div>
+</div>
+
+---
+
+### Loops: cake recipe 
+
+We put an empty bowl on a scale and we keep our ingredients ready.
+
+We need 175g of flour. 
+
+1. What's the action that we need to repeat?
+1. For how long do we need to repeat it?
+
+---
+
+### Loops: cake recipe 
+
+1. The action we need to repeat is adding flour to the bowl
+1. We repeat it as long as the flour in the bowl weights less than 175g
+
+---
+
+### Loops: cake recipe
+
+We can distinguish different parts in a loop:
+
+- a starting point (initialization) 
+- a condition
+- a value that changes to make the condition `false` (update)
+- an action to repeat
+
+It's important that the condition is `false` at some point, otherwise the loop will go on indefinitely.
+
+---
+
+### Loops: cake recipe
+
+- initialization: no flour in the bowl (0g of flour)
+- condition: is the flour weight less than 175g? (flour < 175)
+- update: amount of flour in the bowl (flour++)
+- action: add flour to the bowl
```

Comment:
Just a thought, since we're using actual numbers here, maybe you could say something like "add a measuring cup/spoon of flour (25g)".

Reply:
good point, I'll update it!